### PR TITLE
Encoding

### PR DESCRIPTION
**Pull request by monnerat**
This is an attempt to fix most problems related to encoding and support badly encoded filenames as far as possible.
Subversion is still failing on such filenames (they're causing exceptions in pysvn itself). Git is now able to use them.
This is accomplished by the following new features:
- a new str subclass called S implementing portable unicode/utf8 conversions with surrpogates,
- Duplicating path strings in dialog tables as a displayable field and a hidden unaltered value.
- Implementing workarounds for external library problems (pysvn wants genuine string arguments, Gtk/Gdk fails on invalid characters in sys.argv).

Apart from svn, I have carefully checked possible problems but I probably have missed some of them: comments and reports are welcome!

### CODE REVIEW

File: rabbitvcs/util/strings.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+"""
+
+import sys
+import codecs
+import re
+import six
+import locale
+
+__all__ = ["S", "IDENTITY_ENCODING", "UTF8_ENCODING", "SURROGATEESCAPE"]
```
Comment (adamplumb) [inline rabbitvcs/util/strings.py]: Any reason in particular you didn't put an underscore in here?  Having two Es next to each other like this is a little awkward.

File: rabbitvcs/util/strings.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            if not eh:
+                eh = codecs.lookup_error(errors)
+            p, n = eh(e)
+            output += p
+            if n <= pos:
+                n = pos + 1
+            pos = n
+    return (output, len(input))
+
+class utf8_IncrementalEncoder(codecs.IncrementalEncoder):
```
Comment (adamplumb) [inline rabbitvcs/util/strings.py]: Mixing the underscore and camel case like this seems odd.  Any reason why?

File: rabbitvcs/util/strings.py (hunk 1 above)
Comment (monnerat) [reply rabbitvcs/util/strings.py]: I presume you speak about `SURROGATEESCAPE`? The only reason I did not put an underscore is the standrd Python > 3.1 error handler is called `surrogateescape`. But I'm not stuck on it and if you prefer` SURROGATE_ESCAPE`, I can change it.

File: rabbitvcs/util/strings.py (hunk 2 above)
Comment (monnerat) [reply rabbitvcs/util/strings.py]: None! I generally use underscore, and the suffix matches the base class. I can change it too, if you prefer.

**Comment by monnerat** [comment]
Thanks for the review and approval.

> You might want to check in with some folks who have had codec issues to see if this resolves their issue

I'll probably create an `encoding` label and an `encoding` branch for that purpose.

**Pull request closed without merging**